core/src/main/java/org/frankframework/pipes/EscapePipe.java

```diff
@@ -72,42 +72,71 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			throw new PipeRunException(this, "cannot open stream", e);
 		}
 
-		String substring = null;
-		String result = input;
-		int i = -1;
-		int j = -1;
-		log.debug("substringStart [{}] substringEnd [{}] input [{}]", substringStart, substringEnd, input);
-		if (substringStart != null && substringEnd != null) {
-			i = input.indexOf(substringStart);
-			if (i != -1) {
-				j = input.indexOf(substringEnd, i);
-				if (j != -1) {
-					substring = input.substring(i + substringStart.length(), j);
-					substring = handle(substring);
-					result = input.substring(0, i + substringStart.length()) + substring + input.substring(j);
-				}
-			}
-		} else {
-			result = handle(input);
+		String result = handleSubstrings(input);
+		return new PipeRunResult(getSuccessForward(), result);
+	}
+
+	private String handleSubstrings(String input) {
+		if (substringStart == null || substringEnd == null) {
+			return handle(input);
 		}
 
-		return new PipeRunResult(getSuccessForward(), result);
+		if (substringStart.isEmpty() || substringEnd.isEmpty()) {
+			return input;
+		}
+
+		StringBuilder result = new StringBuilder(input.length());
+		int startIndex = 0;
+
+		while (startIndex < input.length()) {
+			// Find the index of substringStart in the remaining part of the input
+			int start = input.indexOf(substringStart, startIndex);
+
+			// If substringStart is not found, append the remaining part and exit the loop
+			if (start == -1) {
+				result.append(input, startIndex, input.length());
+				break;
+			}
+
+			// Find the index of substringEnd after the current substringStart
+			int end = input.indexOf(substringEnd, start + substringStart.length());
+
+			// If substringEnd is not found, append the remaining part and exit the loop
+			if (end == -1) {
+				result.append(input, startIndex, input.length());
+				break;
+			}
+
+			// Append the part from the last startIndex to the current substringStart
+			result.append(input, startIndex, start + substringStart.length());
+
+			String content = input.substring(start + substringStart.length(), end);
+
+			String handledContent = handle(content);
```

Comment:
I think this cannot be null unless the input is null?

Reply:
It could return null, when the imput XML is not correct this happens